x-pack/plugins/alerting/server/rules_client/tests/bulk_edit.test.ts

```diff
@@ -714,6 +980,16 @@ describe('bulkEdit()', () => {
   });
 
   describe('apiKey operations', () => {
+    beforeEach(() => {
+      mockCreatePointInTimeFinderAsInternalUser({
+        saved_objects: [
```

Comment:
The current code checks that the saved object isn't updated by asserting:
```ts
expect(unsecuredSavedObjectsClient.bulkCreate).toHaveBeenCalledTimes(0);
```
I'm missing the checks for the api keys. Will add it now.

Reply:
Actually, API key regeneration is also done with `unsecuredSavedObjectsClient.bulkCreate`, so the check that I was doing on tests that should skip were enough.

But anyways, I changed the logic so that `bulkMarkApiKeysForInvalidation` is called conditionally on `apiKeysToInvalidate.length > 0`, which makes sense, and also allows to assert separately that:
```ts
expect(bulkMarkApiKeysForInvalidation).toHaveBeenCalledTimes(0);
```
where it corresponds.

Reply:
Ah ok, these checks are included in existing tests like `should skip operation when adding already existing tags`! I missed that. Thanks for adding `expect(bulkMarkApiKeysForInvalidation).toHaveBeenCalledTimes(0);` 🙏